### PR TITLE
Very inspired mech toy suicide

### PR DESCRIPTION
**Pull request by nemvar**
## About The Pull Request

The mech toys now have a unique suicide.

## Why It's Good For The Game

I have to meet my quotas somehow.

## Changelog
:cl:
add: Mech toys now have a unique suicide.
/:cl:

### CODE REVIEW

**Comment by MMMiracles** [comment]
turn "trying to commit suicide" to "preparing for titan fall" because if this isn't just a straight up titanfall reference then i don't even know

**Comment by Dawson1917** [comment]
> user.apply_damage(9999, BRUTE, BODY_ZONE_HEAD)

excessive?

**Comment by fludd12** [comment]
Alternative: `[user] is climbing inside [src]! It looks like they're trying to self-destruct.` after the delay, delete the person, mind transfer them into a mob with the mech's sprite and matching its step noise, and make it harmlessly explode after a few seconds.

**Comment by GuyonBroadway** [comment]
@MMMiracles it could also be in reference to battletechs infamous "death from above" attack.

**Comment by ExcessiveUseOfCobblestone** [comment]
```DM
"[user] looks like they're contemplating whether or not to enter [src] at a crucial point in the story's plot! GET IN THE MECH, [uppertext(user.name)]!"
```

File: code/game/objects/items/toys.dm [hunk 1]
```diff
@@ -448,6 +448,13 @@
 	if(loc == user)
 		attack_self(user)
 
+/obj/item/toy/prize/suicide_act(mob/living/user)
+	user.visible_message("<span class='suicide'>[user] is holding \the [src] above [user.p_them()]self and drops it on [user.p_their()] head! It looks like [user.p_theyre()] trying to commit suicide!</span>")
```
Comment (GuyonBroadway) [inline code/game/objects/items/toys.dm]:
```suggestion
	user.visible_message("<span class='suicide'>[src] fires its jump jets and hovers over [user]'s head! It looks like it's attempting Death From Above!</span>")
```

Comment (GuyonBroadway) [reply code/game/objects/items/toys.dm]: Before you point it out, yes it's daft but ss13 was built on daft, please see the bag of holding suicide, the newspaper suicide and the box suicide for more information.

**Comment by SpaceManiac** [comment]
If you yourself describe it as uninspired, is it really worth adding?

**Comment by PKPenguin321** [comment]
a lot of the suggested messages above me are pretty good so you should instead do a pick() that just does a random one

**Comment by GuyonBroadway** [comment]
Honestly there's a million mech jokes you can make and we'd be here all day if we did them all...


Then again I could do one for each toy in the mech series in a later pr for funsies.

**Comment by nemvar** [comment]
I did random. Fels suggestion was taken.

File: code/datums/components/footstep.dm [hunk 2]
```diff
@@ -14,15 +14,15 @@
 	var/turf/open/T = get_turf(parent)
 	if(!istype(T))
 		return
-
```
Comment (Putnam3145) [inline code/datums/components/footstep.dm]: oof ow ouchie my whitespace

**Comment by ninjanomnom** [comment]
This is dumb and not worth the maintenance costs of the code

**Comment by fludd12** [comment]
> This is ~~dumb~~ fun and ~~not worth the maintenance costs of the code~~ involves an exploding toy mech.

**Comment by Jordie0608** [comment]
Funny, but yeah that code. A mob that exists for 6 seconds just for a suicide act is bad. Especially when you're defined it inside the file for toys ( ͡° ͜ʖ ͡°).

**Comment by nemvar** [comment]
@fludd12 Revive if you want. I'm not wasting any more time on this.

**Pull request closed without merging**